docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,270 @@
+communicate with the database, the frontend service will not run successfully.
+
+### Layer configuration
+
+The web application is initially configured with the Pebble layer below:
+
+```{code-block} yaml
+   
+services:
+  frontend:
+    override: replace
+    command: python3 -m http.server 8080
+    startup: enabled
+  backend:
+    override: replace
+    command: python3 -m http.server 8081
+    startup: enabled
+  database:
+    override: replace
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+```{note}
+We are using Python's http module to mock the servers.
+```
+### Problem with setup
+
+If we start the Pebble daemon (``pebble run``):
+
+```{terminal}
+   :input: pebble run
+
+2024-06-28T02:17:23.347Z [pebble] Started daemon.
+[backend database frontend]
+2024-06-28T02:17:23.353Z [pebble] POST /v1/services 2.613042ms 202
+2024-06-28T02:17:23.356Z [pebble] Service "backend" starting: python3 -m http.server 8081
+2024-06-28T02:17:24.363Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:17:25.370Z [pebble] Service "frontend" starting: python3 -m http.server 8080
+2024-06-28T02:17:26.385Z [pebble] Started default services with change 1.
+```
+
+Ideally we would expect all three services to start up successfully
+(``pebble services``):
+
+```{terminal}
+   :input: pebble services
+
+Service   Startup  Current  Since
+backend   enabled  active   today at 10:17 CST
+database  enabled  active   today at 10:17 CST
+frontend  enabled  active   today at 10:17 CST
+```
+
+However, this configuration does not account for the service dependencies.
+
+For example, the output below shows the ``database`` service failing to start
+("inactive"), but the ``backend`` and ``frontend`` services starting
+successfully ("active"):
+
+```{terminal}
+   :input: pebble run
+
+2024-06-28T02:20:03.337Z [pebble] Started daemon.
+[backend database frontend]
+2024-06-28T02:20:03.343Z [pebble] POST /v1/services 2.763792ms 202
+2024-06-28T02:20:03.346Z [pebble] Service "backend" starting: python3 -m http.server 8081
+2024-06-28T02:20:03.346Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:20:03.347Z [pebble] Service "frontend" starting: python3 -m http.server 8080
+2024-06-28T02:20:03.396Z [pebble] Change 1 task (Start service "database") failed: cannot start service: exited quickly with code 1
+2024-06-28T02:20:04.353Z [pebble] Started default services with change 1.
+```
+
+```{terminal}
+   :input: pebble services
+   
+Service   Startup  Current   Since
+backend   enabled  active    today at 10:20 CST
+database  enabled  inactive  -
+frontend  enabled  active    today at 10:20 CST
+```
+
+We can configure the layer so that Pebble only starts a given service if all
+other services it is dependent on are running successfully to avoid consuming
+resources unnecessarily.
+
+## Define service dependencies
+
+To create service dependencies in Pebble, use the ``requires`` key with
```

Comment:
Why the double backticks? Markdown only requires a single backtick (here and below).

Reply:
Good point @benhoyt ; I get confused switching between RST and MD. :D I'll update the whole doc.